# 1. 라이브러리 임포트

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("Total_Data_1.csv")

# 2. Rule Base일때 광고성 블로그 리뷰를 판단하는지 모델 성능 평가

In [6]:
from sklearn.metrics import accuracy_score, classification_report,confusion_matrix

## 2.1 베이스 모델

In [7]:
print(classification_report(df["광고 분류1"],df["WEB_광고 분류1"]))

              precision    recall  f1-score   support

           0       0.80      0.98      0.88      1980
           1       0.54      0.08      0.15       545

    accuracy                           0.79      2525
   macro avg       0.67      0.53      0.51      2525
weighted avg       0.74      0.79      0.72      2525



In [8]:
print(classification_report(df["광고 분류2"],df["WEB_광고 분류2"]))

              precision    recall  f1-score   support

           0       0.80      0.98      0.88      1980
           1       0.83      0.06      0.11       172
           2       0.42      0.08      0.14       373

    accuracy                           0.78      2525
   macro avg       0.68      0.37      0.38      2525
weighted avg       0.74      0.78      0.72      2525



## 2.2 결론.

광고성 블로그 리뷰를 rule base로 이진 분류를 했을때 정확도는 0.79이고 1에 대한 재현율은 0.08로 낮다. 또한 광고성 리뷰가 아닌 클래스와 쿠팡 파트너스 리뷰와 협찬 등 광고성 블로그 리뷰에 대해서 정확도는 0.78이며, 쿠팡 파트너스에 대한 재현율은 0.06, 광고성 협찬 리뷰에 대해서는 0.08로 매우 낮은 수치이다.

__현 Rule Base로 판단되는 결과가 낮기에 여러가지 전처리 방법과 모델링을 통해 성능 향상을 목표로 진행하고자 합니다.__